scipy/stats/tests/test_hypotests.py

```diff
@@ -1702,39 +1702,54 @@ class TestGamesHowell:
                                  "unequal sample size"])
     def test_compare_spss(self, data, res_expect_str):
         """
+        DATA LIST LIST /Group (F1.0) Value (F8.2).
+        BEGIN DATA
+        0 24
+        0 23
+        0 31
+        0 51
+        1 34
+        1 18
+        1 18
+        1 26
+        2 17
+        2 68
+        2 59
+        2 7
+        END DATA.
+
         ONEWAY Value BY Group
-          /MISSING ANALYSIS
-          /CRITERIA=CILEVEL(0.95)
-          /POSTHOC=GH ALPHA(0.05)
+            /MISSING ANALYSIS
+            /POSTHOC=GH ALPHA(0.5).
         """
         res_expect = np.asarray(
-            res_expect_str.replace(" - ", " ").split()[8:],
+            res_expect_str.replace(" - ", " ").split()[7:],
             dtype=float).reshape(-1, 6)
         res_games = stats.tukey_hsd(*data, equal_var=False)
         conf = res_games.confidence_interval()
         # loop over the comparisons
         for i, j, s, l, h, p in res_expect:
             i, j = int(i), int(j)
-            assert_allclose(conf.low[i, j], l, atol=1e-4)
-            assert_allclose(res_games.statistic[i, j], s, atol=1e-4)
-            assert_allclose(conf.high[i, j], h, atol=1e-4)
-            assert_allclose(res_games.pvalue[i, j], p, atol=1e-3)
+            assert_allclose(res_games.statistic[i, j], s, atol=1e-8)
+            assert_allclose(res_games.pvalue[i, j], p, atol=1e-8)
+            assert_allclose(conf.low[i, j], l, atol=1e-6)
+            assert_allclose(conf.high[i, j], h, atol=1e-5)
 
     r_same_size = """
-               q value Pr(>|q|)  
-    1 - 0 == 0   -1.52  0.55873 
-    2 - 0 == 0    0.48  0.94115
-    2 - 1 == 0    1.25  0.68298  
+                  q value             Pr(>|q|)  
+    1 - 0 == 0   -1.5467805948856344  0.55873362851759
```

Comment:
> Please note that I found our result, R result, and SPSS result vary from each other in high decimal point 

Yup, that's not unexpected. Ours are almost certainly more accurate. When we added the distribution, we studied the accuracy very carefully.

One thing aI wanted to ask about was the changes in this table - do you know why `-1.52` changed to `-1.5467805948856344`? And how do the tests both before and after this changed? (I haven't traced the tests line by line yet to check exactly what is being tested. I will, but can you comment?)

Reply:
> It would be great that you can breifly explain or suggest some reading material or reference

Funny you should ask: [here](https://zenodo.org/records/5151976) is a notebook about implementing the distribution, and [this](https://joss.theoj.org/papers/10.21105/joss.04383) is about `tukey_hsd`.